### PR TITLE
Applab: move debug area below codeTextbox

### PR DESCRIPTION
**Pull request by cpirich**
- Adjust header to keep "block" section visible now that the palette is always up for droplet (code mode and blocks mode)
- Fix headers so the text is not selectable
- Fix exception when toggling blocks quickly (toggleBlocks() won't always return a result)
- Applab: move extraControlRows template (which is essentially the debug area) out of the visualization column and into the area below the codeTextbox div
- Applab: keep the Finish button in the visualization column.
- Applab: make default 'custom' level have default sliderSpeed of 0.95
- page.html.ejs template: move the gameButtons block into a function call.
- Applab: reduce size of droplet palette headers and debug area buttons
  screen shot 2015-03-20 at 9 26 09 am

### CODE REVIEW

File: apps/src/applab/applab.js [hunk 1]
```diff
@@ -580,15 +580,13 @@ Applab.init = function(config) {
...
-  var finishButtonFirstLine = _.isEmpty(level.softButtons) && !showSlider;
   var firstControlsRow = require('./controls.html')({
     assetUrl: studioApp.assetUrl,
     showSlider: showSlider,
-    finishButton: finishButtonFirstLine
+    finishButton: true
   });
   var extraControlsRow = require('./extraControlRows.html')({
     assetUrl: studioApp.assetUrl,
-    finishButton: !finishButtonFirstLine,
```
Comment (Bjvanminnen) [inline apps/src/applab/applab.js]: do we not need to pass in true here?

Comment (Bjvanminnen) [inline apps/src/applab/applab.js]: nm, i see we have it in one place or the other

File: apps/src/templates/page.html.ejs [hunk 2]
```diff
@@ -30,6 +30,26 @@
...
+    <button id="resetButton" class="launch blocklyLaunch" style="display: none">
+      <div><%= msg.resetProgram() %></div>
+      <img src="<%= assetUrl('media/1x1.gif') %>" class="reset26"/>
+    </button>
+    <% if (data.controls) { %>
+    <%- data.controls %>
+    <% } %>
+    <% if (!data.pinWorkspaceToBottom && data.extraControlRows) { %>
+    <%- data.extraControlRows %>
+    <% } %>
```
Comment (Bjvanminnen) [inline apps/src/templates/page.html.ejs]: Is there not a way to do a block of embedded JS instead of each line having <% %>?

**Comment by Bjvanminnen** [comment]
lgtm. might be worth preemptively running eye tests to see if this regresses anything

File: apps/src/applab/extraControlRows.html.ejs [hunk 3]
```diff
@@ -1,31 +1,36 @@
...
-    <button id="stepOutButton" class="share">
-      <%= applabMsg.stepOut() %>
-    </button>
-    <button id="viewDataButton" class="share" style="display:none;">
-      <%= applabMsg.viewData() %>
-    </button>
-    <button id="designModeButton" class="share">
-      <%= applabMsg.designMode() %>
-    </button>
+<div id="debug-area">
```
Comment (davidsbailey) [inline apps/src/applab/extraControlRows.html.ejs]: Please keep this div hidden in Design Mode by adding the following code to Applab.toggleDesignMode:

  var debugArea =  document.getElementById('debug-area');
  debugArea.style.display = enable ? 'none' : 'block';

Comment (cpirich) [inline apps/src/applab/extraControlRows.html.ejs]: I made this change. At the same time, I stopped hiding the gameButtons when you enter Design mode, which we should discuss. I think we may want to leave in the ability to start running even while in design mode.

**Comment by cpirich** [comment]
Ran UI tests locally and everything passed.

File: apps/src/applab/extraControlRows.html.ejs [hunk 4]
```diff
@@ -1,31 +1,36 @@
...
+      <button id="viewDataButton" class="debugger_button" style="display:none;">
+        <%= applabMsg.viewData() %>
+      </button>
+      <button id="designModeButton" class="debugger_button">
+        <%= applabMsg.designMode() %>
+      </button>
+    </div>
+  <% } %>
+
+  <% if (debugButtons) { %>
```
Comment (davidsbailey) [inline apps/src/applab/extraControlRows.html.ejs]: Looks like you can combine this block with the one above it

**Pull request merged**